core/src/main/java/io/questdb/cutlass/http/processors/JsonQueryProcessorState.java

```diff
@@ -507,10 +508,46 @@ private boolean addColumnToOutput(
 
     private void addColumnTypeAndName(RecordMetadata metadata, int i) {
         int columnType = metadata.getColumnType(i);
+        String columnName = metadata.getColumnName(i);
+
+        switch (ColumnType.tagOf(columnType)) {
+            // list of explicitly supported types, to be keep in sync with doQueryRecord()
+
+            // we use a while-list since if we add a new type to QuestDB
+            // the support has to be explicitly added to the JSON REST API
+            case ColumnType.BOOLEAN:
+            case ColumnType.BYTE:
+            case ColumnType.DOUBLE:
+            case ColumnType.FLOAT:
+            case ColumnType.INT:
+            case ColumnType.LONG:
+            case ColumnType.DATE:
+            case ColumnType.TIMESTAMP:
+            case ColumnType.SHORT:
+            case ColumnType.CHAR:
+            case ColumnType.STRING:
+            case ColumnType.VARCHAR:
+            case ColumnType.SYMBOL:
+            case ColumnType.BINARY:
+            case ColumnType.LONG256:
+            case ColumnType.GEOBYTE:
+            case ColumnType.GEOSHORT:
+            case ColumnType.GEOINT:
+            case ColumnType.GEOLONG:
+            case ColumnType.RECORD:
+            case ColumnType.NULL:
+            case ColumnType.UUID:
+            case ColumnType.IPv4:
+            case ColumnType.INTERVAL:
+                break;
+            default:
+                throw CairoException.nonCritical().put("Column type not supported by JSON REST API. Please open a feature request at https://github.com/questdb/questdb/issues/new/ [column=").put(columnName).put(", type=").put(ColumnType.nameOf(columnType)).put(']');
```

Comment:
I'd not suggest opening issue action here TBH, we never do it. Also is we don't support a column type here I'm sure we don't support it in any interface. So maybe this one is better
```suggestion
                throw CairoException.nonCritical().put("column type not supported [column=").put(columnName).put(", type=").put(ColumnType.nameOf(columnType)).put(']');
```